README.md

```diff
@@ -1 +1,7 @@
-# lua-func
\ No newline at end of file
+# lua-func
+
+The lua interpreter split in a way such that each function is in it's own file.
```

Comment:
Every function is in a unique file? I thought some files contained more than one function?

[Also "Lua" with a capital "L" to be polite :)]

Reply:
Done, and yes there are, but I can't seem to find them now

Reply:
So this is why I took so long to respond to you because I was looking for them all morning. But I just remembered the name of the function that caused this problem and now I now which files are like that.

Reply:
Is it just one pair (or more than 2?) functions or ... ? Should we say "all functions are split into single files except X and Y"? Also, can we spell out *why* we've put them in files? Maybe saying something like "This allows us to make each file a shared object and use capabilities to do A, B, and C."?

Reply:
Your change looks good, but can we have a think about:

> Also, can we spell out why we've put them in files? Maybe saying something like "This allows us to make each file a shared object and use capabilities to do A, B, and C."?

Reply:
Instructions on how to build/run/crash this would also be very welcome in the README.

Reply:
The main idea is, not to directly write any capability code and still obtain results. Instructions on building and running should be added yes.

Reply:
OK, let's add some stuff in along those lines. It doesn't have to be super extensive -- a moderate paragraph should be enough by the sounds of things.

Reply:
I have a build script but I am not sure if I should commit it or not: https://gist.github.com/alex4o/2a72535101f078f4b390e8fb1487f5bb

Reply:
I would suggest that you just inline the build script inside the README.md as a code block. It would be difficult to notice the build script amongst the sea of .c files anyway.

Reply:
Is the README fine now?

Reply:
Why wouldn't we include the build script in the repo directly? Maybe even converting it to a Makefile?

Reply:
For clarity what i meant was: "I would suggest, at very minimum inlining the build script". I have no objections to the build script or an equivalent Makefile being in the repo directly.

Given the stated description for the repo, I would imagine this build script is what would allow us to select amongst the "different groupings of functions", eventually.

Reply:
I can use the shell script to generate a Makefile, but that would reduce the understanding of what is actually happaning.